codeSnippets/snippets/auth-oauth-google/src/main/kotlin/com/example/Application.kt

```diff
@@ -0,0 +1,63 @@
+package com.example
+
+import io.ktor.application.*
+import io.ktor.auth.*
+import io.ktor.http.*
+import io.ktor.response.*
+import io.ktor.routing.*
+import io.ktor.client.*
+import io.ktor.client.engine.cio.*
+import io.ktor.client.features.json.*
+import io.ktor.client.request.*
+
+fun Application.main() {
+    val httpClient = HttpClient(CIO) {
+        install(JsonFeature)
+    }
+    install(Authentication) {
+        oauth("auth-oauth-google") {
+            urlProvider = { "http://localhost:8080/callback" }
+            providerLookup = {
+                OAuthServerSettings.OAuth2ServerSettings(
+                    name = "google",
+                    authorizeUrl = "https://accounts.google.com/o/oauth2/auth",
+                    accessTokenUrl = "https://accounts.google.com/o/oauth2/token",
+                    requestMethod = HttpMethod.Post,
+                    clientId = System.getenv("GOOGLE_CLIENT_ID"),
```

Comment:
We should err If environment variable isn't set.

Reply:
I agree that such situations should be handled but I'd not add additional checks here to simplify a code snippet in docs. In addition, an application raises a more or less obvious exception in this case (`System.getenv("GOOGLE_CLIENT_ID") must not be null`).